src/main/java/com/google/cloud/tools/appengine/cloudsdk/CloudSdkAppEngineFlexibleStaging.java

```diff
@@ -60,16 +66,33 @@ public void stageFlexible(StageFlexibleConfiguration config) throws AppEngineExc
           + config.getStagingDirectory().toPath());
     }
 
+    // verification for app.yaml that contains runtime:java
+    Path appYaml = config.getAppEngineDirectory().toPath().resolve("app.yaml");
+    String runtime = "non-determined";
+    try {
+      if (Files.isRegularFile(appYaml)) {
+        runtime = new AppYaml(appYaml).getRuntime();
+      }
+    } catch (IOException e) {
+      log.warning("Unable to determine runtime: error parsing app.yaml");
```

Comment:
You might want to log the actual exception as well to aid debugging. Maybe the file is not readable.